src/Raven.Server/RavenServerStartup.cs

```diff
@@ -277,7 +277,8 @@ private static void CheckDatabaseShutdownAndThrowIfNeeded(RequestHandlerContext
                 return;
 
             var databaseShutdown = context.Database?.DatabaseShutdown ?? context.DatabaseContext?.DatabaseShutdown;
-            if (databaseShutdown is { IsCancellationRequested: true })
+            var databaseShutdownCompleted = context.Database?.DatabaseShutdownCompleted;
```

Comment:
@karmeli87 can you take a look here? I tried to apply this change:

https://github.com/ravendb/ravendb/pull/18167/files#diff-4456fb49c005fe70037bcf8efd8f6404560912c6629cbbaa7ad30a7874a5b888R267

In the sharded database there is no notion of `DatabaseShutdownCompleted`